sa/model.go

```diff
@@ -505,6 +521,42 @@ func authzPBToModel(authz *corepb.Authorization) (*authz2Model, error) {
 	return am, nil
 }
 
+// populateAttemptedFields populates a challenge from a authz2Model with the
+// relevant required fields if it was attempted
```

Comment:
This comment could be improved:
 - The "if it was attempted" suggests that this function checks whether a challenge was attempted or not, but that happens at the higher level. Instead the comment should make clear that this takes an attempted challenge and copies in some fields from the authz2 model.
 - There are few enough fields being copied it makes sense to just list them in the comment.